image_tools/src/showimage.cpp

```diff
@@ -44,12 +44,16 @@ void show_image(const sensor_msgs::msg::Image::ConstSharedPtr & msg) {
   ss << "Received image #" << msg->header.frame_id << std::endl;
   std::cout << ss.str();
 
-  cv::Mat frame(
-    msg->height, msg->width, encoding2mat_type(msg->encoding),
+  // NOTE(esteve): the OpenCV portions of this program are written in C to avoid
+  // the following bug on Windows:
+  // http://stackoverflow.com/questions/20854682/opencv-multiple-unwanted-window-with-garbage-name
+  CvMat frame;
+  cvInitMatHeader(
```

Comment:
same, and I would still keep the creation of a cv::Mat for clarity and to make it easier to migrate one day (to OpenCV 3). cv::Mat constructor with a pointer does not take ownership of the data anyway